src/core/users/users.controller.ts

```diff
@@ -0,0 +1,130 @@
+import {
+  Controller,
+  Get,
+  Post,
+  Body,
+  Patch,
+  Param,
+  Delete,
+  BadRequestException,
+  ConflictException,
+  InternalServerErrorException,
+  NotFoundException,
+  Response,
+} from '@nestjs/common';
+import { UsersService } from './service/users.service';
+import { CreateUserDto } from './dto/create-user.dto';
+import { UpdateUserDto } from './dto/update-user.dto';
+import { ApiTags, ApiOkResponse } from '@nestjs/swagger';
+import { Auth } from '../../core/auth/decorators/auth.decorator';
+import { UserRole } from '../..//constants';
+import { UserCreateResponse } from './dto/create-user-response.dto';
+import { ApiException } from '@nanogiants/nestjs-swagger-api-exception-decorator';
+import { User } from './entities/user.entity';
+import { MessageDTO } from 'src/common/dto/response.dto';
+import * as express from 'express';
+import { USER_SUCCESFULLY_DELETE, USER_SUCCESFULLY_UPDATED } from './messages';
+import { INTERNAL_SERVER_ERROR } from 'src/constants/messages/messagesConst';
+
+@ApiTags('users')
+@Controller('users')
+export class UsersController {
+  constructor(private readonly usersService: UsersService) {}
+
+  @Auth(UserRole.ADMIN)
+  @Post()
+  @ApiOkResponse({
+    status: 200,
+    description: 'Response of user creation',
+    type: UserCreateResponse,
+  })
```

Comment:
Tambien lo ideal seria que retornes un 201 de CREATED y no un 200 de "OK"
PD: esto es opcional, pero es la wena practica

```suggestion
  @ApiCreatedResponse({
    status: 200,
    description: 'Response of user creation',
    type: UserCreateResponse,
  })
```